### PR TITLE
fix(deps): update dependency `@joshdb/provider`

### PR DESCRIPTION
**Pull request by RealShadowNova**

### CODE REVIEW

File: packages/json/src/lib/JSONProvider.ts [hunk 1]
```diff
@@ -88,19 +88,23 @@ export class JSONProvider<StoredValue = unknown> extends JoshProvider<StoredValu
...
 
     if (typeof data !== 'number') {
-      return { ...payload, error: this.error({ identifier: CommonIdentifiers.InvalidDataType, method: Method.Dec }, { key, path, type: 'number' }) };
+      payload.errors.push(this.error({ identifier: CommonIdentifiers.InvalidDataType, method: Method.Dec }, { key, path, type: 'number' }));
+
+      return payload;
     }
 
-    await this[Method.Set]({ method: Method.Set, key, path, value: data - 1 });
+    await this[Method.Set]({ method: Method.Set, errors: [], key, path, value: data - 1 });
```
Comment (dan-online) [inline packages/json/src/lib/JSONProvider.ts:107]: What if set returns a payload with errors?

**Review by dan-online — CHANGES_REQUESTED** [review]
My main question is for every internal call, the initial payload isn't populated with errors

File: packages/json/src/lib/JSONProvider.ts (hunk 1 above)
Comment (RealShadowNova) [reply packages/json/src/lib/JSONProvider.ts:107]: I'd say leave it for this PR, but maybe this should be handled later on with every provider.

Comment (dan-online) [reply packages/json/src/lib/JSONProvider.ts:107]: Perhaps open an issue with high severity, cause it's pretty important

**Review by dan-online — APPROVED** [review]
LGTM

**Pull request merged**